Parsers/ASimNetworkSession/Parsers/ASimNetworkSessionMicrosoftSysmon.yaml

```diff
@@ -0,0 +1,161 @@
+          Initiated:bool,
+          SourceIsIpv6:bool,
+          SourceIp:string,
+          SrcHostname:string,
+          SourcePort:int,
+          SourcePortName:string,
+          DestinationIsIpv6:bool,
+          DestinationIp:string,
+          DestinationHostname:string,
+          DestinationPort:int,
+          DestinationPortName:string
+          ) 
+          with (regex=@'<Data Name="(\w+)">{?([^>]*)}?</Data>')
+         | extend outbound = SourcePort > DestinationPort
+         | project-away EventData
+};
+  let Sysmon3_WindowsEvent=(disabled:bool=false){
+    WindowsEvent
+     | where not(disabled) 
+      | where Provider == "Microsoft-Windows-Sysmon" and EventID == 3
+      | extend
+            RuleName = tostring(EventData.RuleName),
+            UtcTime = todatetime(EventData.UtcTime),
+            ProcessGuid = tostring(EventData.ProcessGuid),
+            ProcessId = tostring(EventData.ProcessId),
+            Image = tostring(EventData.Image),
+            User = tostring(EventData.User),
+            Protocol = tostring(EventData.Protocol),
+            Initiated = tobool(EventData.Initiated),
+            SourceIsIpv6 = tobool(EventData.SourceIsIpv6),
+            SourceIp = tostring(EventData.SourceIp),
+            SrcHostname = tostring(EventData.SrcHostname),
+            SourcePort = toint(EventData.SourcePort),
+            SourcePortName = tostring(EventData.SourcePortName),
+            DestinationIsIpv6 = tobool(EventData.DestinationIsIpv6),
+            DestinationIp = tostring(EventData.DestinationIp),
+            DestinationHostname = tostring(EventData.DestinationHostname),
+            DestinationPort = toint(EventData.DestinationPort),
+            DestinationPortName = tostring(EventData.DestinationPortName)
+    | extend outbound = SourcePort > DestinationPort
+    | project-away EventData
+};
+let OutboundNetworkEvents =  union Sysmon3_Event, Sysmon3_WindowsEvent
+| where outbound
+| extend 
+        SrcUsernameType = 'Simple',
+        SrcUsername = iff(tostring(split(User, "\\")[0]) == Computer, " ", tostring(split(User, "\\")[1])),
+        SrcProcessId = toint(ProcessId), 
+        SrcProcessGuid = tostring(split(split(ProcessGuid, "{")[-1], "}")[0]),
+        SrcProcessName = Image
+| project-away outbound, ProcessId, ProcessGuid, Image
+;
+let InboundNetworkEvents =  union Sysmon3_Event, Sysmon3_WindowsEvent
+| where not(outbound)
+| extend 
+        DstUsernameType = 'Simple',
+        DstUsername = iff(tostring(split(User, "\\")[0]) == Computer, " ", tostring(split(User, "\\")[1])),
+        DstProcessId = toint(ProcessId), 
+        DstProcessGuid = ProcessGuid,
+        DstProcessName = Image
+| project-away outbound, ProcessId, ProcessGuid, Image
+;
+union InboundNetworkEvents,OutboundNetworkEvents
+        | project-rename 
+            EventStartTime = UtcTime,
+            Dvc = Computer,
+            Hostname = DestinationHostname,
+            IpAddr = SourceIp,
```

Comment:
While correct, I feel it is somewhat confusing. I would project-rename to DstIpAddr and SrcIpAddr and have the others as aliases using extend later on.